app/Http/Controllers/Api/MovieController.php

```diff
@@ -15,12 +15,12 @@ class MovieController extends Controller
      */
     public function index()
     {
-        if (request(['offset']) != null and request(['take']) != null ){
-            $data = request(['take', 'offset']);
-            return Movie::offset($data['offset'])->take($data['take'])->get();
-        }else{
-            return Movie::all();
+        if (!empty(request(['page'])) and !empty(request(['perPage'])) ){
+            $data = request(['page', 'perPage']);
+            return Movie::offset($data['page'] * $data['perPage'])->take($data['perPage'])->get();
         }
+        return Movie::all();
```

Comment:
Fali nam ovdje Service, negdje sam to napisao vec, znaci gdje god da nema Service layer-a ubacuj ga